elfo-network/src/discovery/mod.rs

```diff
@@ -126,6 +138,26 @@ impl Discovery {
         Ok(())
     }
 
+    fn make_connections(&mut self) {
+        let (next_check_advise, connections) = self.connman.make_connections();
+        if let Some(advise) = next_check_advise.and_then(|adv| self.advise_timer.feed(adv)) {
+            self.ctx.attach(Delay::until(advise, MakeConnections));
+            info!(
+                after = ?advise.duration_since(Instant::now()),
+                "scheduled next connection opening",
+            );
+        }
+
+        let capabilities = self.get_capabilities();
+        for (prev_state, id) in connections {
+            let conn = self.connman.get_mut(id).unwrap();
+            info!(%conn, %prev_state, "opening connection");
+
+            let stream = Self::open_existing_connection(capabilities, &self.node_map, conn);
```

Comment:
(minor, naming)
We can call it "existing" only if the connection ID remains the same, but it doesn't.